packages/sdk/src/reads/getCurveBestPrice.ts

```diff
@@ -0,0 +1,84 @@
+import { WETH } from "../../tests/constants.js";
+import { ETH_ADDRESS } from "../constants/misc.js";
+import { invariant } from "../utils/assertions.js";
+import { type ReadContractParameters, readContractParameters } from "../utils/viem.js";
+import { type Address, type PublicClient, parseAbi, zeroAddress } from "viem";
+
+const abi = {
+  name: "get_address",
+  outputs: [{ type: "address", name: "" }],
+  inputs: [{ type: "uint256", name: "_id" }],
+  stateMutability: "view",
+  type: "function",
+  gas: 1308,
+} as const;
+
+type Asset = {
+  address: Address;
+  decimals: number;
+  symbol: string;
+};
+
+const curveRegistry = "0x0000000022d53366457f9d5e68ec105046fc4383" as const;
+const swapId = 2n; // id won't change, for swaps it will be always the same id in the registry
+
+export async function getCurveBestPrice(
+  client: PublicClient,
+  args: ReadContractParameters<{
+    incoming: Asset;
+    outgoing: Asset;
+    quantity: bigint;
+  }>,
+) {
+  invariant(args.incoming.address !== args.outgoing.address, "Incoming and outgoing asset are identical");
+
+  const [address] = await client.readContract({
+    ...readContractParameters(args),
+    abi: [abi],
+    functionName: "get_address",
+    address: curveRegistry,
+    args: [swapId],
+  });
+
+  invariant(address !== zeroAddress, "Registry returned zero address");
+  invariant(address !== undefined, "Registry returned undefined address");
+
+  const curveOutgoing = args.outgoing.address === WETH ? ETH_ADDRESS : args.outgoing.address;
+  const curveIncoming = args.incoming.address === WETH ? ETH_ADDRESS : args.incoming.address;
+
+  const curveSwaps = parseAbi([
+    "function get_best_rate(address _from, address to, uint256 amount) view returns (address bestPool, uint256 amountReceived)",
+  ] as const);
+
+  const [bestPool, amountReceived] = await client.readContract({
+    abi: curveSwaps,
+    address: address as Address,
+    functionName: "get_best_rate",
+    args: [curveOutgoing, curveIncoming, args.quantity],
+  });
+
+  invariant(bestPool !== zeroAddress, "Pool returned zero address");
```

Comment:
this is not invariant, invariant are internal checks. Return undefined for example